actions/pull-request/download-artifact/entrypoint/main_test.go

```diff
@@ -0,0 +1,254 @@
+				if req.Method == http.MethodHead {
+					http.Error(w, "NotFound", http.StatusNotFound)
+
+					return
+				}
+
+				switch req.URL.Path {
+				case "/":
+					w.WriteHeader(http.StatusOK)
+
+				case "/repos/some-owner/some-repo/actions/runs/12345/artifacts":
+					filename := "artifact.json"
+					w.Header().Set("Content-Type", "application/json")
+					w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", filename))
+					body, err := os.ReadFile(filename)
+					Expect(err).NotTo(HaveOccurred())
+					_, _ = w.Write(body)
+					w.WriteHeader(http.StatusOK)
+
+				case "/repos/some-owner/some-repo/actions/artifacts/54321/zip":
+					// serving a zip file
+					filename := "payload"
+					buf := new(bytes.Buffer)
+					writer := zip.NewWriter(buf)
+					data, err := ioutil.ReadFile(filename + ".json")
+					if err != nil {
+						log.Fatal(err)
+					}
+					f, err := writer.Create(filename)
+					if err != nil {
+						log.Fatal(err)
+					}
+					_, err = f.Write([]byte(data))
+					if err != nil {
+						log.Fatal(err)
+					}
+					err = writer.Close()
+					if err != nil {
+						log.Fatal(err)
+					}
+					w.Header().Set("Content-Type", "application/zip")
+					w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", filename))
+					_, _ = w.Write(buf.Bytes())
+
+				case "/repos/some-owner/nonexistent-repo/actions/runs/45678/artifacts":
+					w.WriteHeader(http.StatusNotFound)
+
+				case "/repos/some-owner/some-repo/actions/artifacts/55555/zip":
+					filename := "another-payload"
+					w.Header().Set("Content-Type", "application/json")
+					w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", filename))
+					w.WriteHeader(http.StatusOK)
+
+				default:
+					t.Fatal(fmt.Sprintf("unknown path: %s", req.URL.Path))
+				}
+			}))
+
+			tempDir, err = os.MkdirTemp("", "output")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(os.Setenv("GITHUB_WORKSPACE", tempDir)).To(Succeed())
+		})
+
+		it.After(func() {
+			mockServer.Close()
+			Expect(os.Unsetenv("GITHUB_WORKSPACE")).To(Succeed())
+			Expect(os.RemoveAll(tempDir)).To(Succeed())
+		})
+
+		context("given arguments that point to a repo with an artifact", func() {
+			it("gets the artifact zip and unzips it onto the file system", func() {
+				command := exec.Command(
+					entrypoint,
+					"--name", "payload",
+					"--repo", "some-owner/some-repo",
+					"--runID", "12345",
+					"--githubAPI", mockServer.URL,
+				)
+
+				buffer := gbytes.NewBuffer()
+				session, err := gexec.Start(command, buffer, buffer)
+				Expect(err).NotTo(HaveOccurred())
+
+				Eventually(session).Should(gexec.Exit(0), func() string { return string(buffer.Contents()) })
+
+				out := string(buffer.Contents())
+				Expect(out).To(ContainLines(
+					"Getting workflow artifacts",
+					"Getting workflow artifact zip file",
+					"Reading file: payload",
+				))
```

Comment:
```suggestion
				Expect(buffer).To(gbytes.Say("Getting workflow artifacts"))
				Expect(buffer).To(gbytes.Say("Getting workflow artifact zip file"))
				Expect(buffer).To(gbytes.Say("Reading file: payload"))
```

It might be worth considering using [`gbytes.Say`](https://pkg.go.dev/github.com/onsi/gomega/gbytes#Say) in this case. It works natively with a `gbytes.Buffer`. It also supports RegEx natively and has some other cool features. It may save you from pulling in `occam` as a dependency entirely.

Reply:
this is so much better YES! thank you

Reply:
fixed in https://github.com/paketo-buildpacks/github-config/pull/267/commits/c99cf4683950a4fd2371e9c5d89d2c25e0ec9974